### ISSUE TITLE
The async Logstash client does not support the "ttl" option

### ISSUE THREAD

**Issue by mbudge** [elastic/beats#39127]
Hi,

Any idea when TTL will be supported with the async client?

"The ttl option is not yet supported on an asynchronous Logstash client (one with the pipelining option set)"

https://www.elastic.co/guide/en/fleet/current/logstash-output.html

Thanks

**Comment by cmacknz** [comment]
The documentation on the TTL option from https://www.elastic.co/guide/en/beats/filebeat/current/logstash-output.html#_ttl:

> Time to live for a connection to Logstash after which the connection will be re-established. Useful when Logstash hosts represent load balancers. Since the connections to Logstash hosts are sticky, operating behind load balancers can lead to uneven load distribution between the instances. Specifying a TTL on the connection allows to achieve equal connection distribution between the instances. Specifying a TTL of 0 will disable this feature.
> 
> The default value is 0. This setting accepts [duration](https://www.elastic.co/guide/en/beats/libbeat/8.13/config-file-format-type.html#_duration) data type values.

**Comment by nimarezainia** [comment]
Yeah apologies. @mbudge wrote that also in the description, totally missed it.

**Comment by ar3diu** [comment]
So... it's just an error message and the TTL should be applied even if pipelining is enabled?

**Comment by cmacknz** [comment]
The TTL setting is ignored and does nothing when in async mode. In sync mode it works as described.

**Comment by ar3diu** [comment]
> The TTL setting is ignored and does nothing when in async mode. In sync mode it works as described.

And how should the async mode be disabled?
`pipelining: 0`?

**Comment by ar3diu** [comment]
I tried to set the following for the Logstash output in Fleet, but I still get some errors in regards to agent connections to the Logstash hosts. Any idea what could cause them?

```
pipelining: 0
ttl: 30s
```

```
error closing connection to logstash host <redacted>: tls: failed to send closeNotify alert (but connection was closed anyway): write tcp <redacted>-><redacted>: wsasend: An established connection was aborted by the software in your host machine., reconnecting...
```

**Comment by daniel-vasek** [comment]
I think the answer has not been answered. Setting the TTL is the only option how to evenly distribute the load through the load balancer (btw the default way of load balancing in Kubernetes). Setting pipelining to 0 reduces the throughput of the logging pipeline significantly, so the possibility of using TTL with pipelining would be super useful for those who need to deliver huge number of logs.

**Comment by michbeck100** [comment]
When will `ttl` be available for async clients? Disabling pipelining seems to be not desireable.
